wither/src/model.rs

```diff
@@ -399,7 +398,9 @@ fn build_index_map(list_index: Document) -> HashMap<String, IndexModel> {
                     options.insert(b_key.to_string(), b_value);
                 }
             });
-            let model = IndexModel::new(idx_keys.clone(), Some(options));
+
+            let options: IndexOptions = mongodb::bson::from_document(options).unwrap(); // FIXME: throw error
```

Comment:
Yea, given that this is now a fallible operation, we should change this function overall to be of a fallible interface. That will allow us to remove the unwrap, and just propagate the bson error directly.

Reply:
Changed this function to propagate the bson error. This is fixed as of now.